src/main/java/net/ripe/ipresource/IpResource.java

```diff
@@ -133,12 +133,16 @@ public final int hashCode() {
         } else if (value.contains(this)) {
             return Collections.emptyList();
         } else {
-            List<IpResourceRange> result = new ArrayList<IpResourceRange>(2);
-            if (getStart().compareTo(value.getStart()) < 0) {
-                result.add(getStart().upTo(value.getStart().predecessor()));
+            final List<IpResourceRange> result = new ArrayList<IpResourceRange>(2);
+            final UniqueIpResource start = getStart();
```

Comment:
It doesn't make a lot of sense, since complier (or JIT) always optimises these repetitive calls out.

Reply:
The compiler cannot optimize this (`getStart`/`getEnd`) may have side-effects, so it uses an "expensive" `invokevirtual` to load the value each time. The JIT might only optimize this after >10k invocations (by default).

That being said, I'm not sure this optimization is worth it here. It's all quite cheap operations.

Reply:
Sure? Optimising out getters is like the most important optimisation in Java ever :)
I have no solid knowledge here, so if these changes increase the probability of this 
thing being faster, I would do it.

Reply:
For `invokestatic` the compiler can do some optimizations, but with `invokevirtual` it definitely cannot -- as that would require making assumptions about unknown implementations. The crux is that these "getter" methods are abstract.